x-pack/test/alerting_api_integration/security_and_spaces/group2/tests/actions/sub_action_framework/index.ts

```diff
@@ -0,0 +1,297 @@
+          });
+
+          const { message, ...resWithoutMessage } = execRes.body;
+          expect(resWithoutMessage).to.eql({
+            status: 'error',
+            retry: false,
+            connector_id: res.body.id,
+          });
+        }
+      });
+
+      it('should execute correctly without schema validation', async () => {
+        const res = await createSubActionConnector({ supertest });
+        objectRemover.add('default', res.body.id, 'action', 'actions');
+
+        const execRes = await executeSubAction({
+          supertest,
+          connectorId: res.body.id as string,
+          subAction: 'subActionWithoutParams',
+          subActionParams: {},
+        });
+
+        expect(execRes.body).to.eql({
+          status: 'ok',
+          data: { id: 'test' },
+          connector_id: res.body.id,
+        });
+      });
+
+      it('should return an empty object if the func returns undefined', async () => {
+        const res = await createSubActionConnector({ supertest });
+        objectRemover.add('default', res.body.id, 'action', 'actions');
+
+        const execRes = await executeSubAction({
+          supertest,
+          connectorId: res.body.id as string,
+          subAction: 'noData',
+          subActionParams: {},
+        });
+
+        expect(execRes.body).to.eql({
+          status: 'ok',
+          data: {},
+          connector_id: res.body.id,
+        });
+      });
+
+      it('should return an error if sub action is not registered', async () => {
+        const res = await createSubActionConnector({ supertest });
+        objectRemover.add('default', res.body.id, 'action', 'actions');
+
+        const execRes = await executeSubAction({
+          supertest,
+          connectorId: res.body.id as string,
+          subAction: 'notRegistered',
+          subActionParams: { foo: 'foo' },
+        });
+
+        expect(execRes.body).to.eql({
+          status: 'error',
+          message: 'an error occurred while running the action',
+          retry: false,
+          connector_id: res.body.id,
+          service_message: 'Sub action not registered',
+        });
+      });
+
+      it('should return an error if the registered method is not a function', async () => {
+        const res = await createSubActionConnector({ supertest });
+        objectRemover.add('default', res.body.id, 'action', 'actions');
+
+        const execRes = await executeSubAction({
+          supertest,
+          connectorId: res.body.id as string,
+          subAction: 'notAFunction',
+          subActionParams: { foo: 'foo' },
+        });
+
+        expect(execRes.body).to.eql({
+          status: 'error',
+          message: 'an error occurred while running the action',
+          retry: false,
+          connector_id: res.body.id,
+          service_message: 'Not valid method for registered sub action',
+        });
+      });
+
+      it('should return an error if the registered method is not a function', async () => {
```

Comment:
```suggestion
      it('should return an error if the requested method is not registered', async () => {
```

Actually, maybe this is only testing the case of a subaction connector registered, but it has no subactions?  That's what it looks like from the service message.